sdk/apis/apis/v1alpha2/types_apibinding_conversion.go

```diff
@@ -0,0 +1,203 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"encoding/json"
+	"fmt"
+
+	kubeconversion "k8s.io/apimachinery/pkg/conversion"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/sdk/apis/apis/v1alpha1"
+)
+
+const (
+	AcceptablePermissionClaimsAnnotation = "apis.v1alpha2.kcp.io/acceptable-permission-claims"
+)
+
+// v1alpha2 -> v1alpha1
+
+func Convert_v1alpha2_APIBinding_To_v1alpha1_APIBinding(in *APIBinding, out *apisv1alpha1.APIBinding, s kubeconversion.Scope) error {
+	out.ObjectMeta = *in.ObjectMeta.DeepCopy()
+
+	_, overhangingAPC, err := Convert_v1alpha2_AcceptablePermissionClaims_To_v1alpha1_AcceptablePermissionClaims(in.Spec.PermissionClaims, s)
+	if err != nil {
+		return err
+	}
+	if len(overhangingAPC) > 0 {
+		encoded, err := json.Marshal(overhangingAPC)
+		if err != nil {
+			return fmt.Errorf("failed to encode claims as JSON: %w", err)
+		}
+
+		if out.Annotations == nil {
+			out.Annotations = map[string]string{}
+		}
+		out.Annotations[AcceptablePermissionClaimsAnnotation] = string(encoded)
+	}
+
+	if err := Convert_v1alpha2_APIBindingSpec_To_v1alpha1_APIBindingSpec(&in.Spec, &out.Spec, s); err != nil {
+		return err
+	}
+
+	if originalPermissionClaims, ok := in.Annotations[PermissionClaimsV1Alpha1Annotation]; ok {
+		permissionClaims := []apisv1alpha1.AcceptablePermissionClaim{}
+		if err := json.Unmarshal([]byte(originalPermissionClaims), &permissionClaims); err != nil {
+			return fmt.Errorf("failed to decode schemas from JSON: %w", err)
+		}
+
+		for _, pc := range permissionClaims {
+			for i, opc := range out.Spec.PermissionClaims {
+				if pc.PermissionClaim.EqualGRI(opc.PermissionClaim) {
+					out.Spec.PermissionClaims[i].ResourceSelector = pc.ResourceSelector
+				}
+			}
+		}
+
+		delete(out.Annotations, PermissionClaimsV1Alpha1Annotation)
+
+		// make tests for equality easier to write by turning []string into nil
+		if len(out.Annotations) == 0 {
+			out.Annotations = nil
+		}
+	}
+
+	return Convert_v1alpha2_APIBindingStatus_To_v1alpha1_APIBindingStatus(&in.Status, &out.Status, s)
+}
+
+func Convert_v1alpha2_AcceptablePermissionClaims_To_v1alpha1_AcceptablePermissionClaims(in []AcceptablePermissionClaim, s kubeconversion.Scope) (out []apisv1alpha1.AcceptablePermissionClaim, overhanging []AcceptablePermissionClaim, err error) {
+	for _, apc := range in {
+		if len(apc.PermissionClaim.Verbs) == 1 && apc.PermissionClaim.Verbs[0] == "*" {
+			var v1pc apisv1alpha1.PermissionClaim
+			err := Convert_v1alpha2_PermissionClaim_To_v1alpha1_PermissionClaim(&apc.PermissionClaim, &v1pc, s)
```

Comment:
Why is this not calling `Convert_v1alpha2_AcceptablePermissionClaim_To_v1alpha1_AcceptablePermissionClaim` as this is about `AcceptablePermissionClaims`?

Reply:
Updated.